checks.d/consul.py

```diff
@@ -0,0 +1,215 @@
+# 3p
+import requests
+
+# stdlib
+import time
+
+# project
+from checks import AgentCheck
+
+
+class ConsulCheck(AgentCheck):
+    CONSUL_CHECK = 'consul.up'
+    HEALTH_CHECK = 'consul.check'
+
+    CONSUL_CATALOG_CHECK = 'consul.catalog'
+    CONSUL_NODE_CHECK = 'consul.node'
+
+    def __init__(self, name, init_config, agentConfig, instances=None):
+        AgentCheck.__init__(self, name, init_config, agentConfig, instances)
+
+        self._last_known_leader = None
+
+    def consul_request(self, instance, endpoint):
+        url = "{0}{1}".format(instance.get('url'), endpoint)
+        try:
+            resp = requests.get(url)
+        except requests.exceptions.Timeout:
+            self.log.exception('Consul request to {0} timed out'.format(url))
+            raise
+
+        resp.raise_for_status()
+        return resp.json()
+
+    ### Consul Config Accessors
+    def _get_local_config(self, instance):
+        #TODO: Maybe store the config on `self` here so we don't have to keep retrieving it?
```

Comment:
We should implement your todo if it makes us save one http call per iteration. You could invalidate periodically, so store the timestamp it was last fetched and invalidate every 5min.